app.rb

```diff
@@ -1,10 +1,67 @@
 require 'sinatra'
 require 'active_record'
 require 'sqlite3'
+require 'pry'
+require_relative 'models/note'
+require_relative 'models/tagging'
+require_relative 'models/tag'
 
 ActiveRecord::Base.establish_connection(
   adapter:  'sqlite3',
   database: 'db/development.sqlite3'
 )
 
+get '/api/notes.json' do
+  notes = Note.all
+  notes.to_json
+end
 
+get "/api/notes/tag/:name" do
+  tag = Tag.find_by(name: params['name'])
+  if tag != nil
+    json = tag.to_json
+  else
+    status 400
```

Comment:
This is probably more of a straight 404 ... and you don't have a `note` variable in here regardless.